

syntax = "proto3";
package jdfl;

message Status {
  int32 status = 1;
  string error_message = 2;
}

message Request {
  repeated string uuid = 1;  // AppID_Service_index
  string ip_port = 2;        // ip:port
}

message ServicePair {
  string local_uuid = 1;
  string remote_uuid = 2;
}

message PairInfoResponse {
  Status  status = 1;
  repeated ServicePair service_pair = 2;
}

service PairService {
  rpc RegisterUUID(Request) returns (Status);
  rpc GetPairInfo(Request) returns (PairInfoResponse);
}